src/api/app/views/webui2/webui/repositories/_repository_entry.html.haml

```diff
@@ -3,12 +3,11 @@
     .card-header
       = link_to(repository, project_repository_state_path(project: project, repository: repository.name), class: 'font-weight-bold')
       .small
-        - if repository.architectures.size.zero?
+        - if repository.architectures.blank?
```

Comment:
doing the right thing would be a change for the good :)

Reply:
I think `blank?` is here the right thing (not sure if that is what @coolo meant with his lass comment). If it is preloaded, then both are equivalent, but if not `blank?` will load it while `empty?` will not. Taking into account you are using it in line 9 (always that you can sue the catched result), **`blank?` is the right thing**. :nerd_face:

Reply:
I think the benefits outweight the few ns it takes Rails to decide if it should use option A or option B.
And this code will stay for a while. It might well be that the surrounding code changes and the option suddenly is faster.

Reply:
I understand @bgeuken  - we should rather promote the use of the smart `empty?` than expecting everyone in the team to understand when to use what (that fails too often :speak_no_evil:)